### PR TITLE
 Enable coverage checks in CI for java8

### PR DESCRIPTION
**Pull request by mar-kolya**
This enabled coverage checks in CI.
Unfortunately this is not really bullet proof, for example checks will not be performed if there are no tests, e.g.: mongo-async-3.3.

### CODE REVIEW

File: dd-java-agent/instrumentation/lettuce-5/src/test/groovy/LettuceSyncClientTest.groovy [hunk 1]
```diff
@@ -42,6 +42,8 @@ class LettuceSyncClientTest extends AgentTestRunner {
   @Shared
   RedisClient redisClient = RedisClient.create(EMBEDDED_DB_URI)
 
+  @Shared
+  StatefulConnection connection
```
Comment (gary-huang) [inline dd-java-agent/instrumentation/lettuce-5/src/test/groovy/LettuceSyncClientTest.groovy]: Good catch. I can add this to the other test suites too if you want? (LettuceReactiveClientTest, LettuceAsyncClientTest)

Comment (mar-kolya) [reply dd-java-agent/instrumentation/lettuce-5/src/test/groovy/LettuceSyncClientTest.groovy]: Sorry, didn't cache it before. Updated now.

**Comment by mar-kolya** [comment]
> Looks good. I'm good to merge if coverage reports (jacocoTestReport) are picked up in the test artifacts?

https://5560-89221572-gh.circle-artifacts.com/0/home/circleci/dd-trace-java/reports/dd-java-agent/instrumentation/spymemcached-2.12/jacoco/datadog.trace.instrumentation.spymemcached/index.html - they are being picked up, but they do look weird, so I'm not sure this is very useful

File: dd-java-agent/instrumentation/lettuce-5/src/test/groovy/LettuceSyncClientTest.groovy (hunk 1 above)
Comment (gary-huang) [reply dd-java-agent/instrumentation/lettuce-5/src/test/groovy/LettuceSyncClientTest.groovy]: Thanks! No need to apologize, I should be the one apologizing. :)

**Comment by tylerbenson** [comment]
Code coverage was working at some point before.  I think that was broken when I split up the job for java 7 and 9.  Can we enable it for them too?  Thanks for fixing this again.

**Pull request merged**